### ISSUE TITLE
section[["SA", "byStation"]] should work

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1392]
At the moment, `[[,section-method` ignores the second argument (if provided), but I think if we say e.g. `section[["SA", "byStation"]]` then we should get a list in which each item equal the SA values computed for the associated station.

Below is the present code. (PS. it is not terribly efficient, since it uses `c()`, but that is only called `n` times, if `n` is the number of stations in a section, and that means likely only a few hundred times).

Since the present behaviour ignores the second argument to `[[`, I should be able to make my change without breaking anything, and will likely code this sometime before the upcoming Saturday, unless a co-developer objects.

```R
              } else if ("SA" == i) {
                  res <- NULL
                  for (station in x[["station"]])
                      res <- c(res, station[["SA"]])
```

**Comment by dankelley** [comment]
Fixed in "dk" branch commit c20f0b9ac400b43d013caedb9b75398085d6f979.

I plan to merge "dk" into "develop" on Sat March 31.

The fix includes the following build-time test.
```R
test_that("section[[...]] and [[..., \"byStation\"]] work", {
          data(section)
          for (i in c("CT", "depth", "nitrate", "nitrite", "oxygen",
                      "phosphate", "potential temperature", "pressure", "SA",
                      "salinity", "sigmaTheta", "silicate", "spice",
                      "temperature", "theta", "z")) {
            v <- section[[i]]
            expect_true(is.vector(v))
            expect_equal(length(v), 2841)
            l <- section[[i, "byStation"]]
            expect_true(is.list(l))
            expect_equal(head(v, 5), l[[1]])
            expect_equal(length(l), 124)
          }
})
```

**Comment by dankelley** [comment]
"dk" merged to "develop" after extensive testing with local "test" and "check" in RStudio and then
```R
rhub::check(env_vars = c("_R_CHECK_FORCE_SUGGESTS_" = "false"))
```
(and then answering to use a windows release build), plus other Rhub checks. (This env-var setting requirement seems to be new.)
